designs/tripleo/osc-tripleo.md

```diff
@@ -0,0 +1,137 @@
+*OSC In the TripleO Overcloud*  
+
+## Constraints and Assumptions
+### OSC Packaged Binaries
+TripleO can only deploy OSC through a template if the OSC binaries are available in a public repository. There are multiple paths to achieve this:  
+1.  RPM package.
+2.  Docker container image on an arbitrary Docker repository.
+3.  Docker container image on a Kolla repository.
+
+The source of the OSC binaries is outside of the scope for this document but it assumes option *3. Kolla repository* at the moment. For any other choice the OSC THT may need changes and/or additional Puppet manifests might be needed.   
+
+### Puppet TripleO
+No changes or new files will be needed in the Puppet TripleO Repository: [https://github.com/openstack/puppet-tripleo](https://github.com/openstack/puppet-tripleo)   
+
+## Design Changes
+A THT describes the OpenStack deployment in Heat Orchestration Template YAML files and Puppet manifests. A template for the OSC service will be created and registered with TripleO. The goal is for all the changes described on this section to be upstreamed to the [THT repository](#tripleo-repo)
+
+### OSC THT  
+
+#### Template Location  
+The OSC THT will be  `/tripleo-heat-templates/docker/services/osc.yaml`. The resource must be registered in the environment file `tripleo-heat-templates/environments/docker.yaml` as follows:  
+```OS::TripleO::Services::OSC: ../docker/services/osc.yaml```  
+
+With these files in place the user can include the OSC service as a resource in the *overcloud* by adding `OS::TripleO::Services::OSC` to the *controller* role during deployment and providing *--environment* files with the values of the parameters defined on the template.   
+
+#### Template Content
+The file `../docker/services/osc.yaml` starts with the template version and a description, followed a set of parameters, a list of resources and outputs generated by the template. For details on THTs, see the [guide](#tht-guide).  
+
+```yaml  
+heat_template_version: rocky
+
+description: >
+  Open Security Controller: service for orchestration of virtual network security functions.
+
+ parameters:
+  #  The paramater EndpointMap is required for all service templates*.
+  EndpointMap:
+    default: {}
+    description: Mapping of service endpoint -> protocol. Typically set
+                 via parameter_defaults in the resource registry.
+				 
+  # The name of the OSC image in the Kolla repository**.
+  OSCDockerImage:
+    description: image
+    type: string
+
+  # Placeholder to exemplify an OSC environment variable.
+  OSCEnvVar:
+    description: image
+    type: string
+	
+resources:
+  # This resource containers a static list of configurations and scripts necessary 
+  # for the deployment of container services in the *overcloud*
+  ContainersCommon:
+    type: ./containers-common.yaml
+
+outputs:
+  # Service templates must output a role_data value containing the config_settings and step_config for the service configuration**
```

Comment:
I do not understand? can we add description what is role data? or point to relevant link, which helps us to understand better?

Reply:
This is also a required output for all TripleO services. It outputs value settings used to configure the services on the overcloud. The doc has a reference for that here, thus the `*` (was **, typo) in the end of the sentence. [THT Guide](https://docs.openstack.org/heat/pike/template_guide/hot_guide.html)